### DISCUSSION TITLE
onMutate and onSuccess of useConnect are not triggered when using autoconnect

### DISCUSSION THREAD

**Discussion by antoinefarley** [wevm/wagmi#2937] [Q&A] [answered]
When using autoconnect: true in createConfig, the onMutate and onSuccess callbacks of useConnect are not triggered when accounts are loaded automatically (i.e. on first render). Is that normal behaviour and if so, is there an alternative solution? Thanks!

**Comment by SaadNiaziDev** [comment, marked as answer]
You are using wrong hook for your usecase. 
https://wagmi.sh/react/hooks/useAccount#onconnect-optional
You have to handle isReconnected bit in onConnect callback to see whether it reconnected or freshly connected
